src/content/en/fundamentals/engage-and-retain/push-notifications/subscribing-a-user.md

```diff
@@ -0,0 +1,396 @@
+
+You only need to create these keys once for your application, just make sure you keep the
+private key private....yeah I just said that.
+
+## Permisions and subscribe()
+
+There is one side effect of calling `subscribe()`. If your web app doesn't have permissions for
+showing notifications at the time of calling `subscribe()`, the browser will request the
+permissions for you. This is useful if your UI works with this flow, but if you want more
+control (and I think most developers will), stick to the `Notification.requestPermission()` API
+that we used earlier.
+
+## What is a PushSubscription?
+
+We call `subscribe()`, pass in some options, in return we get a promise that resolves to a
+`PushSubscription` resulting in some code like so:
+
+    function subscribeUserToPush() {
+      return getSWRegistration()
+      .then(function(registration) {
+        const subscribeOptions = {
+          userVisibleOnly: true,
+          applicationServerKey: urlBase64ToUint8Array(
+            'BEl62iUYgUivxIkv69yViEuiBIa-Ib9-SkvMeAtA3LFgDzkrxZJjSgSnfckjBJuBkr3qBUYIHBQFLXYp5Nksh8U'
+          )
+        };
+
+        return registration.pushManager.subscribe(subscribeOptions);
+      })
+      .then(function(pushSubscription) {
+        console.log('Received PushSubscription: ', JSON.stringify(pushSubscription));
+        return pushSubscription;
+      });
+    }
+
+The `PushSubscription` object contains all the required information needed to send a push
+messages to that user. If you print out the contents using `JSON.stringify()`, you'll see the
+following:
+
+    {
+      "endpoint": "https://some.pushservice.com/something-unique",
+      "keys": {
+        "p256dh":
+    "BIPUL12DLfytvTajnryr2PRdAgXS3HGKiLqndGcJGabyhHheJYlNGCeXl1dn18gSJ1WAkAPIxr4gK0_dQds4yiI=",
+        "auth":"FPssNDTKnInHVndSTdbKFw=="
+      }
+    }
+
+The `endpoint` is the push services URL, you make a POST request to this URL to trigger a push
+message.
+
+The `keys` object contains the values used to encrypt message data sent with a push message
+(which we'll discuss later on in this book).
+
+## Send a Subscription to Your Server
+
+Once you have a push subscription you'll want to send it to your server. It's up to you how you
+do that but a tiny tip is to use `JSON.strinigify()` to get all the necessary data out of the
+subscription object, otherwise you can piece together the same result manually:
+
+    const subscriptionObject = {
+      endpoint: pushSubscription.endpoint,
+      keys: {
+        p256dh: pushSubscription.getKeys('p256dh'),
+        auth: pushSubscription.getKeys('auth')
+      }
+    };
+
+    // The above is the same output as:
+
+    const subscriptionObjectToo = JSON.stringify(pushSubscription);
+
+In [the demo used throughout this
+book](https://github.com/gauntface/web-push-book/tree/master/src/demos/node-server), we make a
+POST request to send a subscription to our node server that stores the subscription in a
+database.
```

Comment:
I'm going to remove this as it doesn't make much sense in the context of web fundamentals.